### PR TITLE
Dont re-xmit received gossip

### PR DESCRIPTION
**Pull request by rustyrussell**
Closes: #2990

### CODE REVIEW

File: common/per_peer_state.c [hunk 1]
```diff
@@ -118,6 +120,7 @@ bool time_to_next_gossip(const struct per_peer_state *pps,
 		*t = time_from_sec(0);
 	else
 		*t = timemono_between(pps->gs->next_gossip, now);
+
```
Comment (cdecker) [inline common/per_peer_state.c]: Spurious whitespace change.

File: common/per_peer_state.c [hunk 2]
```diff
@@ -107,8 +110,7 @@ static inline bool timemono_after(struct timemono a, struct timemono b)
 	return time_greater_(a.ts, b.ts);
 }
 
-bool time_to_next_gossip(const struct per_peer_state *pps,
-			 struct timerel *t)
+bool time_to_next_gossip(const struct per_peer_state *pps, struct timerel *t)
```
Comment (cdecker) [inline common/per_peer_state.c]: Whitespace only change.

**Comment by rustyrussell** [comment]
It's 10,000 allocs, so that's closer to 10000 x 48 bytes.  We could make it much more efficient with a simple bump allocator, but in practice if you send us 10k updates within 60 seconds, either we're initially syncing, or you're spamming.

**Comment by rustyrussell** [comment]
Trivial rebase

**Comment by rustyrussell** [comment]
OK, since Travis found a (not really but) memleak, I had to do some memleak infra work.  But the world is a nicer place now...

**Comment by cdecker** [comment]
It seems `test_node_reannounce` now is really flaky or broken.

**Comment by rustyrussell** [comment]
Hmm, will investigate today.

**Pull request merged**